bundles/org.openhab.binding.cbus/src/main/java/org/openhab/binding/cbus/internal/CBusThreadPool.java

```diff
@@ -45,10 +45,10 @@ protected synchronized CGateThreadPoolExecutor CreateExecutor(@Nullable String n
     }
 
     public class CBusThreadPoolExecutor extends CGateThreadPoolExecutor {
-        private final ThreadPoolExecutor threadPool;
+        private final ExecutorService threadPool;
 
         public CBusThreadPoolExecutor(@Nullable String poolName) {
-            threadPool = (ThreadPoolExecutor) ThreadPoolManager.getPool("binding.cbus-" + poolName);
+            threadPool = (ExecutorService) ThreadPoolManager.getPool("binding.cbus-" + poolName);
```

Comment:
When looking at the PR that introduced this problem for this binding, openhab/openhab-core#2531, I'm wondering:

Perhaps `getPoolUnwrapped` can be called instead of `getPool` directly? It returns `ThreadPoolExecutor`, so maybe could be done even without any casting. If this would be possible, this problem can not be introduced as easily again, because it would generate compile-time error instead of runtime exception.